src/main/haskell/kore/src/Kore/Implicit/ImplicitVarsInternal.hs

```diff
@@ -48,5 +51,5 @@ v2 = implicitUnparameterizedVariable "#v2"
 vx = implicitUnparameterizedVariable "#x"
 vx' = implicitUnparameterizedVariable "#x'"
 
-pS = sortParameter Meta "#sp" AstLocationImplicit
+pS = sortParameter (Proxy::Proxy Meta) "#sp" AstLocationImplicit
```

Comment:
`Proxy` is `Applicative`, so that
```.hs
Proxy :: Proxy Meta
===
pure Meta
```
I'm not saying you need to change it here, I'm just mentioning it. It is especially convenient when a function is parametric in the level, so you write something like,
```.hs
foo :: MetaOrObject level => level -> Proxy level
foo level = pure level
```
instead of using `ScopedTypeVariables` (which requires writing an explicit `forall` and generally gets messy).

General question: Is there a reason you are using a proxy (`Proxy level`) instead of a witness (`IsMetaOrObject level`)?

Reply:
Why Proxy?  I needed a function of the form
```
sortParam :: String -> SortVariable level
sortParam name = sortParameter Proxy name AstLocationTest
```
Here, the parameter `level` is on the result sort, so I thought `Proxy` would perform the job better.
Would you think of a better solution in this context?